jmclient/jmclient/wallet_service.py

```diff
@@ -805,12 +805,22 @@ def sync_addresses(self):
 
     def sync_unspent(self):
         st = time.time()
-        # block height needs to be real time for addition to our utxos:
-        current_blockheight = self.bci.get_current_block_height()
+        # block height needs to be real time for addition to our utxos.
+        # We are a little aggressive in trying this RPC call, because if
+        # it fails we cannot continue:
+        for _ in range(5):
+            current_blockheight = self.bci.get_current_block_height()
+            if current_blockheight:
+                break
```

Comment:
Should we add some delay here maybe? In case bitcoind is temporary not responding, five very fast consecutive calls has higher probability to fail than, let's say, five calls in five seconds.

Reply:
Not a bad idea but I'm worried about doing it synchronously.
There is already a delay where you wait for the RPC call to return (that part of the code is synchronous now).

Reply:
See https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1348/commits/84686b96a7cc785288b0522a66481e99a75e44fb . I thought about this for a while and decided: this is a step before the main monitoring loop begins, and it is by design a synchronous operation; I'm not actually sure an extra half second sleep is needed (since we are in any case waiting for the RPC call to return), but, just in case it helps, I added it.